clang/lib/Tooling/Inclusions/Stdlib/StandardLibrary.cpp

```diff
@@ -115,15 +115,17 @@ static int initialize(Lang Language) {
       NSLen = 0;
     }
 
-    if (SymIndex >= 0 &&
-        Mapping->SymbolNames[SymIndex].qualifiedName() == QName) {
-      // Not a new symbol, use the same index.
+    if (SymIndex > 0) {
       assert(llvm::none_of(llvm::ArrayRef(Mapping->SymbolNames, SymIndex),
```

Comment:
This assertion could fail only if entries for a symbol were partially grouped, but it didn't fail if entries weren't grouped at all.
Example:
```cpp
// StdSpecialSymbolMap.inc
SYMBOL(any_cast, std::, <any>)
SYMBOL(div, std::, <cstdlib>)
// StdSymbolMap.inc
SYMBOL(any, std::, <any>)
SYMBOL(any_cast, std::, <any>)  // assertion doesn't fail because previous entry has a different name
...

```

Reply:
this assertion is making sure we're inserting all the headers for a symbol consecutively. this is important for correctness as we want to know all the headers providing a symbol not a random set.

we should fix the mappings files instead to not split `any_cast` (and other symbols).

Reply:
I've already fixed the mappings, but I'd like to keep the changes in assertion to simplify debugging if someone makes the same mistake in the future.

When I first tried to run the generator and didn't remove the duplicate mappings like `any_cast` from `StdSpecialSymbolMap.inc` (d19c4aa7a6e58ef7870343f4f663417edf06189e), I got a segfault which was quite hard to debug (see description of #113494).

Rebuilding clangd with `-DLLVM_ENABLE_ASSERTIONS=ON` didn't help, because the assertion didn't check the second mapping for `any_cast` (because of `Mapping->SymbolNames[SymIndex].qualifiedName() == QName` condition).
This change makes the assertion fail for such unordered mappings.